### PR TITLE
4.6

### PR DESCRIPTION
**Pull request by vkarpov15**
Fix issues from #4341

### CODE REVIEW

**Comment by vkarpov15** [comment]
Gonna hold off on this for now and investigate more later why the tests are flaking. Sorry @donnanicolas gonna have to hold off on releasing this...

**Comment by donnanicolas** [comment]
@vkarpov15 ok, we will continue using our fork. Tell me if there is any way I can help with

**Comment by donnanicolas** [comment]
@vkarpov15 any updates on this??

**Comment by vkarpov15** [comment]
Mongodb driver 2.2.6 is out so I'm gonna push a release with that early this week

File: test/model.findOneAndUpdate.test.js [hunk 1]
```diff
@@ -909,7 +909,7 @@ describe('model: findByIdAndUpdate:', function() {
     });
   });
 
-  it('can do deep equals on object id after findOneAndUpdate (gh-2070)', function(done) {
+  it.only('can do deep equals on object id after findOneAndUpdate (gh-2070)', function(done) {
```
Comment (donnanicolas) [inline test/model.findOneAndUpdate.test.js]: I don't know it this `it.only(...)` is deliberated or not ... ❓

Comment (vkarpov15) [inline test/model.findOneAndUpdate.test.js]: Not deliberate, removed it in next commit, good catch :+1:

**Pull request merged**